app/src/main/java/io/gnosis/safe/ui/settings/app/SettingsHandler.kt

```diff
@@ -2,15 +2,31 @@ package io.gnosis.safe.ui.settings.app
 
 import android.view.Window
 import android.view.WindowManager
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.appcompat.app.AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+import androidx.appcompat.app.AppCompatDelegate.NightMode
 import pm.gnosis.svalinn.common.PreferencesManager
 import pm.gnosis.svalinn.common.utils.edit
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class PrivacySettingsHandler @Inject constructor(
+class SettingsHandler @Inject constructor(
     private val preferencesManager: PreferencesManager
 ) {
+    @NightMode
+    var nightMode: Int
+        get() = preferencesManager.prefs.getInt(KEY_NIGHT_MODE, MODE_NIGHT_FOLLOW_SYSTEM)
+        set(value) {
+            preferencesManager.prefs.edit {
+                putInt(KEY_NIGHT_MODE, value)
+            }
+        }
+
+    fun applyNightMode(@NightMode nightMode: Int) {
```

Comment:
can we combine setting nightMode and applying it into one step (move apply to set(value) of nightMode). They are used always together anyway

Reply:
In BaseActivity.onCreate() it is not applied via the setter:
`settingsHandler.applyNightMode(settingsHandler.nightMode)`